airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -699,11 +701,16 @@ def _merge_stream_slicers(self, model: DeclarativeStreamModel, config: Config) -
                 ),
                 partition_router=stream_slicer,
             )
+        elif not model.incremental_sync and stream_slicer:
```

Comment:
see other comment, but due to `model.incremental_sync and stream_slicer`, if we get to this point and we have a `stream_slicer` then we know that there is no `model.incremental_sync` present

Reply:
Uh, I see it now, the old 
```python
elif stream_slicer: 
    return stream_slicer 
```

condition is not needed now because we covered all the cases where we want the streams to use the `per_partition_cursors,` the only case left is when we don't have the `stream_slicer,`, and thus, we will use the `ResumableFullRefreshCursor` else `return None.` 

Corrected.